src/ir.jl

```diff
@@ -1,3 +1,46 @@
+using YaoHIR: X, Z, S, T, SWAP, Rz, Rx, shift
+
+convert_to_gate(::Val{:X}, loc) = Gate(X, Locations(loc))
```

Comment:
if we just `XGate` type over `Val{:X}` we can prob get rid of these functions in this package.

Reply:
It will simplify for most gates except for `CNOT` and `CZ`. For consistency, make it's better to use the old one.